src/common/Themes/theme_helpers.cpp

```diff
@@ -0,0 +1,111 @@
+// Port Based on https://stackoverflow.com/a/62811758/5001796
+#include "theme_helpers.h"
+#include "dwmapi.h"
+#include <windows.h>
+#include <iostream>
+#include <vector>
+
+#define STATUS_SUCCESS 0x00000000
+#define DWMWA_USE_IMMERSIVE_DARK_MODE 20
+#define HKEY_WINDOWS_THEME L"Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize"
+
+typedef NTSTATUS(WINAPI* RtlGetVersionPtr)(PRTL_OSVERSIONINFOW);
+
+namespace
+{
+    RTL_OSVERSIONINFOW GetRealOSVersion()
+    {
+        HMODULE hMod = ::GetModuleHandleW(L"ntdll.dll");
+        if (hMod)
+        {
+            RtlGetVersionPtr fxPtr = (RtlGetVersionPtr)::GetProcAddress(hMod, "RtlGetVersion");
+            if (fxPtr != nullptr)
+            {
+                RTL_OSVERSIONINFOW rovi = { 0 };
+                rovi.dwOSVersionInfoSize = sizeof(rovi);
+                if (STATUS_SUCCESS == fxPtr(&rovi))
+                {
+                    return rovi;
+                }
+            }
+        }
+        RTL_OSVERSIONINFOW info = { 0 };
+        return info;
+    }
+
+    DWORD WINAPI CheckImmersiveDarkMode(LPVOID lpParam)
```

Comment:
where is this used?

Reply:
Whoops! This should have been removed.
Good catch!